debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -418,6 +424,23 @@ private Scn getMaxArchiveLogScn(List<LogFile> logFiles) {
             }
         }
 
+        if (currentRedoLogFiles.isEmpty()) {
+            throw new DebeziumException("Cannot get current redo log SCN as no redo logs are present.");
```

Comment:
AFAICT `currentRedoLogFiles` is initialized only when archive mode is on and thus `isStartScnInArchiveLogs()` is called, so if we don't use archive mode, we would fail here?

Reply:
I'm really concerned about the use of `currentRedoLogFiles`, even beyond Vojtech's comments above.  This is a collection we calculate as a one-off in `isStartScnInArchiveLogs`, and its safe to use those because we're trying to determine where the SCN in question is in the archive and redo logs.

However, calculating the max SCN should be based solely on the logs that are part of the mining session, which is calculated elsewhere when we populated the `currentLogFiles` collection, and I think ultimately that should be the method call that we base this explicit calculation upon so that we don't run the risk of any type of race condition with data sets.

Looking at the code more closely, I'm inclined to say that the call to `setLogFilesForMining` when archive log mode is enabled should return all the current logs (archive and redo). This will allow the collection of the current redo log sequences to be calculated properly. Then you could use that full collection here in this calculation without relying on the `currentRedoLogFiles` collection that is calculated elsewhere based on a one-off query that could be based on a completely different state.

If you two concur, then I would suggest the following changes:

1. Introduce a new class called `LogFiles` or similar, and it acts as a parameter object, and it will expose two critical pieces of information (a) all log files for a given SCN, (b) all log files that are to be mined.
2. `LogMinerHelper#getLogFilesForOffsetScn` would return this `LogFiles` object, populated accordingly.
3. `LogMinerHelper#setLogFilesForMining` will return this `LogFiles` object.

This will then allow us inside this method to have access to both ALL and Mined log files, and we can use the ALL collection here in this method so that the list of logs is consistent with those that were added to the session without there being any risk of de-synced data sets.